crates/nargo_cli/src/cli/fs/common_reference_string.rs

```diff
@@ -19,9 +24,24 @@ fn common_reference_string_location() -> PathBuf {
 pub(crate) fn read_cached_common_reference_string() -> Vec<u8> {
     let crs_path = common_reference_string_location();
 
-    // TODO(#1390): Implement checksum
     match std::fs::read(crs_path) {
-        Ok(common_reference_string) => common_reference_string,
+        Ok(common_reference_string) => {
+            let loaded_checksum = CRS_CHECKSUM.load(std::sync::atomic::Ordering::Acquire);
+            let checksum = CASTAGNOLI.checksum(&common_reference_string);
+            if loaded_checksum == 0 {
+                CRS_CHECKSUM
+                    .compare_exchange_weak(
+                        loaded_checksum,
+                        checksum,
+                        std::sync::atomic::Ordering::Acquire,
+                        std::sync::atomic::Ordering::Relaxed,
+                    )
+                    .unwrap();
+            } else if checksum != loaded_checksum {
+                panic!("Common reference string checksum mismatch");
```

Comment:
I believe we'd want to return an empty vector so the CRS will be fetched again since it was somehow corrupted.